### PR TITLE
[release-1.5] Manual cherrypick  - Add annotations for setting cpu/memory limits on sidecar 

### PR DESCRIPTION
**Pull request by stewartbutler**
Manual cherrypick of #22395

CC @rolandkool

### CODE REVIEW

File: mixer/pkg/protobuf/yaml/testdata/all/types.pb.go [hunk 1]
```diff
@@ -10391,6 +10391,7 @@ func (m *Outer_Inner) Unmarshal(dAtA []byte) error {
 func skipTypes(dAtA []byte) (n int, err error) {
 	l := len(dAtA)
 	iNdEx := 0
+	depth := 0
```
Comment (howardjohn) [inline mixer/pkg/protobuf/yaml/testdata/all/types.pb.go]: Something went wrong here, I would just revert the changes to the mixer stuff?

Comment (stewartbutler) [reply mixer/pkg/protobuf/yaml/testdata/all/types.pb.go]: That was produced by make gen, I assumed because one of the base charts was changed. 

I can revert it, though.

Comment (howardjohn) [reply mixer/pkg/protobuf/yaml/testdata/all/types.pb.go]: I think it must be some different dep you have locally or something?

**Comment by stewartbutler** [comment]
Replaced by https://github.com/istio/istio/pull/22468
